### ISSUE TITLE
Explanation about the mechanism of model forward function

### ISSUE THREAD

**Issue by sh0416** [facebookresearch/llama#209]
Hi

I just wondering how the `self.model.forward(tokens[:, prev_pos:cur:pos], prev_pos)` would be sufficient to generate the logits for next word.

I think it could come from the property of relative positional encoding, but I couldn't figure out why.

Is there anyone who can explain about this mechanism?

Thank you

**Comment by sh0416** [comment]
Duplicated: https://github.com/facebookresearch/llama/issues/151
